src/heitech.zer0mqXt.core/Patterns/Retry.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Threading.Tasks;
+using heitech.zer0mqXt.core.infrastructure;
+
+namespace heitech.zer0mqXt.core.patterns
+{
+    ///<summary>
+    /// Encapsulates the retry policy grasped from the configuration
+    ///</summary>
+    internal class Retry
+    {
+        private readonly SocketConfiguration _configuration;
+
+        internal Retry(SocketConfiguration configuration)
+        {
+            this._configuration = configuration;
+        }
+
+        ///<summary>
+        /// Register your retryable callback here
+        ///</summary>
+        internal async Task<XtResult<T>> RunAsyncWithRetry<T>(Func<Task<XtResult<T>>> retryableAction, string operationName = "request")
+            where T : class
+        {
+            try
+            {
+                return await retryableAction().ConfigureAwait(false);
+            }
+            catch (NetMQ.EndpointNotFoundException ntfnd)
+            {
+                _configuration.Logger.Log(new ErrorLogMsg($"NetMQ.Endpoint could not be found at {_configuration.Address()}: " + ntfnd.Message));
+                await Task.Delay((int)_configuration.TimeOut.TotalMilliseconds).ConfigureAwait(false);
```

Comment:
`Task Delay` also has a `TimeSpan` overload: [Task.Delay(TimeSpan)](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.delay#System_Threading_Tasks_Task_Delay_System_TimeSpan_).
Let's either use that overload, or change the `IZer0MqBuilder.SetTimeOut(long timeOutInMs)` method to take an `int` instead, since the consumer may end up with unwanted overflows here, or even an [OverflowException](https://docs.microsoft.com/en-us/dotnet/api/system.overflowexception) if they have enabled `<CheckForOverflowUnderflow>true</CheckForOverflowUnderflow>` in their projects. ⚠️

Reply:
Ah ok, did not know that. I will fix that immediately

Reply:
done